Add /api/virtualization/virtual-machines/{id}/render-config/ endpoint

pkorovin opened the pull request:
<!--
    Thank you for your interest in contributing to NetBox! Please note that
    our contribution policy requires that a feature request or bug report be
    approved and assigned prior to opening a pull request. This helps avoid
    waste time and effort on a proposed change that we might not be able to
    accept.

    IF YOUR PULL REQUEST DOES NOT REFERENCE AN ISSUE WHICH HAS BEEN ASSIGNED
    TO YOU, IT WILL BE CLOSED AUTOMATICALLY.

    Please specify your assigned issue number on the line below.
-->
### Fixes: #13808

<!--
    Add /api/virtualization/virtual-machines/{id}/render-config/ endpoint
-->

pkorovin committed:
Add /api/virtualization/virtual-machines/{id}/render-config/ endpoint

LuPo commented on netbox/virtualization/api/views.py:
```diff
@@ -78,6 +83,23 @@ def get_serializer_class(self):
+    @action(detail=True, methods=['post'], url_path='render-config', renderer_classes=[JSONRenderer, TextRenderer])
+    def render_config(self, request, pk):
+        """
+        Resolve and render the preferred ConfigTemplate for this Device.
```
Docstring should mention "..this Virtual Machine" instead of "...this Device"

pkorovin committed:
Update Docstring "Device" -> "Virtual Machine"

Docstring should mention "..this Virtual Machine" instead of "...this Device", thanks @LuPo!

jeremystretch committed:
Move config rendering logic to new RenderConfigMixin

jeremystretch commented:
Thanks @pkorovin! I've refactored this a bit to avoid duplicating the rendering logic.

jeremystretch committed:
Add tests for render-config API endpoint
